low-level/sensors/impl/generic/node_hw.py

```diff
@@ -480,6 +481,63 @@ def run(self):
             logger.warning(f"{self.SENSOR_NAME} Node hw monitoring disabled")
             self.shutdown()
 
+    def _check_faulty_resource_status(self):
+        # Currently, the SSPL sensor relies on IPMI SEL events to track server
+        # resources fault & recovery. However, its been observed multiple times
+        # with Supermicro servers, that SEL not getting updated in case recoveries
+        # of some resources.
+        # here we are polling resources that are detected to be faulty or missing
+        # periodically, in addition to current polling for SEL.
+
+        # Copying object to avoid RuntimeError: dictionary changed size during iteration
+        faulty_res = self.faulty_resources.copy()
+        for sensor_id in faulty_res:
+            dynamic, static = self._get_sensor_sdr_props(sensor_id)
+            if dynamic and 'States Asserted' in dynamic:
+                #  'States Asserted': 'Power Supply, Presence detected'
+                resource_state = re.sub(',  +', ', ', re.sub('[\[\]]','',
+                    dynamic['States Asserted']).replace('\n',','))
+                sensor_status = resource_state.split(',')
+                sensor_status = sensor_status[0] if len(sensor_status) == 1 \
+                    else sensor_status[1]
+                sensor_status = sensor_status.strip()
+                if sensor_status in ["", "Presence detected"]:
+                    self._generate_resource_alert(sensor_id, faulty_res)
+            elif static and 'Assertion Events' in static \
+                and 'Status' in static:
+                # 'Status': 'ok'
+                # 'Assertion Events': ''
+                if static['Assertion Events'] in [""] \
+                    and static['Status'] == 'ok':
+                    self._generate_resource_alert(sensor_id, faulty_res)
+        del faulty_res
+
+    def _generate_resource_alert(self, sensor_id, faulty_res):
+        status = faulty_res[sensor_id]['status'].lower()
+        status = 'Deasserted' if status == 'asserted' else 'Asserted'
```

Comment:
>status = 'Deasserted' if status == 'asserted'

Why status is inverted here,  to 'Deasserted' or vice versa?

Reply:
Any fault resolved resources status is inverted of faulty resources or vice versa.
Example: (Voltage)
**8 | 05/17/2021 | 19:23:31 | Power Supply #0xc4 | Failure detected () | Asserted**
9 | 05/17/2021 | 19:23:32 | Fan #0x45 | Lower Critical going low  | Asserted
**a | 05/17/2021 | 19:29:25 | Power Supply #0xc4 | Failure detected () | Deasserted**

Addressed this review comments in https://github.com/Seagate/cortx-monitor/pull/393